arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -64,7 +64,7 @@ static void _region_init(u32_t index, u32_t region_addr,
 	/* Select the region you want to access */
 	ARM_MPU_DEV->rnr = index;
 	/* Configure the region */
-	ARM_MPU_DEV->rbar = region_addr | REGION_VALID | index;
+	ARM_MPU_DEV->rbar = region_addr & REGION_ADDR_MASK;
```

Comment:
> Even if I agree with the solution it does not seem to me to be related to the "mpu_stack_guard_test" example.

You're right. This issue is not related to the "mpu_stack_guard_test" example.
My original intention is to provide an instant example to show what would happen if we don't mask the last 5 bits of the RBAR.
I've modified the commit message to prevent confusing.

> Could you please change this line as per below?
ARM_MPU_DEV->rbar = (region_addr & REGION_ADDR_MASK) | REGION_VALID | index;

Done.